src/views/award.ts

```diff
@@ -70,6 +84,119 @@ export class AwardView extends BlockView<Trackable<AwardNode>> {
     element.textContent = textContent
     return element
   }
+
+  handleClick = () => {
+    if (isDeleted(this.node) || !this.props.getCapabilities().editArticle) {
+      return
+    }
+    this.showContextMenu()
+  }
+
+  showContextMenu = () => {
+    this.props.popper.destroy()
+    const componentProps: ContextMenuProps = {
+      actions: [
+        {
+          label: 'Edit',
+          action: () => {
+            this.props.popper.destroy()
+            this.showAwardModal(this.node)
+          },
+          icon: 'Edit',
+        },
+        {
+          label: 'Delete',
+          action: () => {
+            this.props.popper.destroy()
+            this.showDeleteAwardDialog()
+          },
+          icon: 'Delete',
+        },
+      ],
+    }
+    this.props.popper.show(
+      this.dom,
+      ReactSubView(
+        this.props,
+        ContextMenu,
+        componentProps,
+        this.node,
+        this.getPos,
+        this.view,
+        'context-menu'
+      ),
+      'right-start',
+      false
+    )
+  }
+
+  showAwardModal = (award?: AwardNode) => {
+    this.dialog?.remove()
+    this.popperContainer?.remove()
+
+    const componentProps: AwardModalProps = {
+      initialData: award?.attrs || ({} as AwardAttrs),
+      onSaveAward: this.handleSaveAward,
+      onCancelAward: this.handleCancelAward,
+    }
+    this.popperContainer = ReactSubView(
+      this.props,
+      AwardModal,
+      componentProps,
+      this.node,
+      this.getPos,
+      this.view,
+      'award-editor'
+    )
+    this.props.popper.show(this.dom, this.popperContainer, 'auto', false)
+  }
+
+  handleSaveAward = (award: AwardAttrs) => {
+    updateNodeAttrs(this.view, schema.nodes.award, award)
+  }
+
+  handleCancelAward = () => {
+    // TODO: Inserted award should be removed on cancel
+  }
+
+  showDeleteAwardDialog = () => {
+    this.dialog?.remove()
+
+    const award = this.node
+    const pos = this.getPos()
+
+    const handleDelete = () => {
+      if (award) {
+        const tr = this.view.state.tr
+        const from = pos // Start position of the node
+        const to = pos + award.nodeSize // End position of the node
+        this.view.dispatch(tr.delete(from, to))
+      }
+    }
+
+    const componentProps: DeleteAwardDialogProps = {
+      handleDelete: handleDelete,
+    }
+
+    this.popperContainer = ReactSubView(
+      this.props,
+      DeleteAwardDialog,
+      componentProps,
+      this.node,
+      this.getPos,
+      this.view,
+      'award-editor'
+    )
+    this.props.popper.show(this.dom, this.popperContainer, 'auto', false)
+  }
+
+  public selectNode = () => {
```

Comment:
We can make this a method, and call `super.selectNode()` instead of updating the classList.

Reply:
right